ImportC fails when DMD is installed in path with spaces

## elpenguino+D reported this on 2024-07-31T15:37:35Z

### Transferred from https://issues.dlang.org/show_bug.cgi?id=24692

### Description

````markdown
Sample code:

hello.c
```
int main() { return 0; }
```

When DMD is installed to a path with spaces in it, an error similar to the following is produced:
```
hello.c: fatal error C1083: Cannot open include file: 'C:\Program': No such file or directory
c1: fatal error C1083: Cannot open source file: 'Files\D\dmd2\windows\bin64\..\..\src\druntime\import\importc.h': No such file or directory
```
````

--- a/compiler/src/dmd/link.d
+++ b/compiler/src/dmd/link.d
@@ -993,7 +993,7 @@ public int runPreprocessor(Loc loc, const(char)[] cpp, const(char)[] filename, c
                  */
                 OutBuffer buf;
                 buf.writestring(cpp);
-                buf.printf(" /P /Zc:preprocessor /PD /nologo /utf-8 %.*s /FI%s /Fi%.*s",
+                buf.printf(" /P /Zc:preprocessor /PD /nologo /utf-8 \"%.*s\" \"/FI%s\" \"/Fi%.*s\"",
                     cast(int)filename.length, filename.ptr, importc_h, cast(int)output.length, output.ptr);
 
                 /* Append preprocessor switches to command line

